### DISCUSSION TITLE
Position Zimm2 2/3/4 Aug 2006 - Position / Residuals show major variation

### DISCUSSION THREAD

**Discussion by Bemtgenj** [groops-devs/groops#74] [Q&A]
Hello,

I am running Luxembourg stations from 2006 to 2021 and to confirm that my script works properly I have used Zimm2 as a baseline. When I processed / plotted the position and the residuals for 2/3/4 Aug 2006 i saw some bands with lots of noise on the residual plots (between 11:00 and 18:00 and hardly any for the other periods) as well as major position shifts for the station on those days (see plots). As Zimm2 is a very stable station we did not expect those results and discussing it here in the office we think it might have something to do with the Ionosphere model. Have you seen this type of behavior / output before? If yes do you know where it comes from? Do we need to change some variables in the Ionosphere model to cover this issue, or is it coming from somewhere else. BTW i see the same pattern on the Luxembourg station for those days.
Thank you very much for your help
Cheers
Jean Bemtgen
[zimmAug2006.zip](https://github.com/groops-devs/groops/files/9187310/zimmAug2006.zip)

**Comment by Waley04** [comment]
Hi Jean,

Before the GROOPS developers respond. 

I tried to process for this station using the following specifications:

**==> Products** :  TUG0R03FIN_*
**==> System**    :  GPS

Here are my solutions: 

1). **position_2006-08-02 zimm**
![position_2006-08-02 zimm](https://user-images.githubusercontent.com/101952777/181085698-b81882ef-8612-4062-bc6e-b82babc482bd.png)

2). **residuals_2006-08-02 zimm**
![residuals_2006-08-02 zimm](https://user-images.githubusercontent.com/101952777/181085816-b33e904e-a2ed-4598-ad3d-7853cc38df82.png)

I am not sure about the differences in our solutions. May I know the products which you used so that I can also verify the stability in the time series. 

Best wishes,

Waley.

**Comment by Bemtgenj** [comment]
Hi Waley

TX for the quick feedback and running the script (i just saw that i had a typo in the station name - the station is zimm and not zim2) , These plots look more of what i expected the graphs to look like like.  I have attached the files I am using (i added a few days to the zip file).
When you state system: GPS - what exactly do you mean with that in the sense did you change something in the scripts?

f you need further info please let me know
Cheers
Jean
[Data.zip](https://github.com/groops-devs/groops/files/9195958/Data.zip)

**Comment by Bemtgenj** [comment]
Addendum: would it be possible for you to send me the scripts you used then i could compare and see if there are some differences between your scripts and mine unless it comes from the products side

TXS
Cheers
Jean

**Comment by Bemtgenj** [comment]
Addendum 2: System: GPS - did you mean you just used GPS as the only constellation in the script or did you mean more changes?
Cheers
Jean

**Comment by Bemtgenj** [comment]
In the processing script - GNSS section i took out the 2 lines for Glonass and Galileo and reran it / plotted the outcome - same as before 

![positionGPS_2006-08-02 zimm](https://user-images.githubusercontent.com/100558965/181209420-dd76c505-ab2c-4c30-873e-5f2bccbd84f8.png)

![residualsGPS_2006-08-02 zimm](https://user-images.githubusercontent.com/100558965/181209456-14bfc811-431b-4070-a2ce-0d353bbbfe76.png)

Cheers
Jean

**Comment by Bemtgenj** [comment]
The only change i have done in the base script is setting the elevation angle to 15Deg. are there any other changes you have done?

**Comment by akvas** [comment]
Dear @Bemtgenj and @Waley04,

we decided to set up GitHub discussions for usage and processing related questions. The motivation behind this step was that discussions remain visible, while issues are hard to keep track of once they are closed. I encourage you to continue this thread, or post your next question in the [Q&A section](https://github.com/groops-devs/groops/discussions/categories/q-a) of our discussion board. Thank you!

Cheers,
Andreas

**Comment by Bemtgenj** [comment]
Hello,

we have investigated this item further and think our difference in results for 2006 might come from the Vienna mapping file as this is teh only change to the base script. As the the GROOPS baseline database did not have the 2006 vmf i created those by downloading the necessary files form the Vienna University and then as per Sebastian use the  following program in the script: ViennamappingFinctionGrid2File  

@Waley04 
In addition to maybe sending me a copy of you script to compare with mine would it also be possible to send me the Viennamapping for Aug 2006 for comparison - unless you used the gpt model?

Tank you vert much for feedback,

BR
Jean

**Comment by Waley04** [comment]
Hi Jean,

I noted that the problem is with the way you converted the VMF. I get similar results between GPT and VMF. Unfortunately, I used a different PC to conduct such an exercise which I am, currently, not having. To solve this, I will rewrite the VMF modeling script and send you together with the PPP script. Alternatively, you may post the script you used, then I will check if it matches my original script.

I hope this is in order.

Regards,

Waley.

**Comment by Bemtgenj** [comment]
Hi Waley

Txs for the quick feedback.

this is the vmf script i am using

[Vmf3.zip](https://github.com/groops-devs/groops/files/9233683/Vmf3.zip)

I was going to send you my aug 2006 vmf file too, but the file in its zip form is too large to upload (about 160MB)

i downloaded the input files from this link (https://vmf.geo.tuwien.ac.at/trop_products/GRID/1x1/V3GR/V3GR_EI/) for 2006 data

Cheers
Jean

**Comment by Bemtgenj** [comment]
and for completeness 
my groops processing scripts (same file as above in first point
[zimmAug2006.zip](https://github.com/groops-devs/groops/files/9233708/zimmAug2006.zip)
)

**Comment by Waley04** [comment]
Hi Jean,

Here is my script for converting raw VFM data to groops format:

**===> convertVmf3groopsFormat.xml**
[convertVmf3groopsFormat.zip](https://github.com/groops-devs/groops/files/9237072/convertVmf3groopsFormat.zip)

Just put the VMF data in a directory and label it **rawVmfDataset**.  Note this is just my choice. In this script, you may use any folder names in lines 6 and 7 - to store the raw and converted data, respectively.  

**===> You may edit the folder names**
![vmfDirectories](https://user-images.githubusercontent.com/101952777/182240072-6f632a4c-c8ed-4e04-8293-5d30755a1101.PNG)

For instance, in my case, I have 
**1). convertVmf3groopsFormat.xml**
**2). rawVmfDataset**

I just run the command to get the **convertedVMF** directory. 

I hope this helps.

Cheers.

**Comment by Bemtgenj** [comment]
Hi Waley

thank you very much for the script.

 I noticed in the script you send me the input file had a *.txt extension, do you do any preprocessing?

So i downloaded the 2006 V3GR files from the following link products https://vmf.geo.tuwien.ac.at/trop_products/GRID/1x1/V3GR/V3GR_EI/). Added them to the directories and ran the script (i did not see a difference to the one i had run except that is use for 2006 vmf3 files as input to the script), then processed / plotted the data and got the following output
![image](https://user-images.githubusercontent.com/100558965/182361646-4b5e86b8-c584-4dc1-8245-731bc7ce5564.png)
 I have still a lot of noise around the11:00-1;00 and around 6;00pm periods

Do you think you can send me your processing script maybe there are some differences in the base script and the aug 2008 vienna mapping file so that i really use your input file (as the upload will not be possible you can send it to the following email address jean.bemtgen@uni.lu). I think i really need to compare everything now as we still have different results for the same day.

On other item we noticed is that if we run processing with the gpt model option we do get similar noisy results, would this mean it is not vienna mapping and as no vienna mapping files are directly used in this run?

@Sebastian, @Barbara,- is it also possible to get the Aug 2006 Vienna mapping GROOPS file (as it is not on the server)  so that i can run the script with that too and compare the data.

**Comment by Bemtgenj** [comment]
ii am planning to lay side by side the following graphs: the 2 different converting script based plots for aug 2 and also next to it the graphs with vmf and3 and V3GR input files - so that we have all of them on the same page

Cheers
Jean

**Comment by Waley04** [comment]
Hi Jean,

I also grabbed the data from [here](https://vmf.geo.tuwien.ac.at/trop_products/GRID/1x1/V3GR/V3GR_EI/), the same link. I would attach my converted VMF file. However, it is larger than the acceptable size. You would try to process using the data I converted. 

I still suspect the VMF file you converted. If I may ask, how big in MB is your converted VMF file? 

Regards,

Waley.

**Comment by Bemtgenj** [comment]
HI Waley

My Aug 2006 vienna mapping file has as a zipped file 239897 kb and as an unzipped file 506764 kB - i had notices before that the zipped files had different size in a 2020 comparison to the Groops baseline files but the unzipped versions had the same size.

Please disregards the above plot as i made a mistake in the script - i was pointing towards the wrong file

you can send the files directly to me email address - jean.bemtgen@uni.lu - no problem

Cheers
Jean

**Comment by Bemtgenj** [comment]
Hi Waley

Parisa and i ran the following scenarios for zimm 2Aug 2006: your script / V3GR input, your script / VMF 3 input and then GROOPS based script / V3GR, GROOPS based script / VMF3 and processing with gpt model (see attached PowerPoint) - i noticed that although the unzipped files are the same size the zipped files have a different size between the vmf3 and v3gr input files
[Zimm4Plots.pptx](https://github.com/groops-devs/groops/files/9243222/Zimm4Plots.pptx)

**Comment by Bemtgenj** [comment]
bottom right has a typo on page 1 it should state GROOPS

**Comment by Bemtgenj** [comment]
updated version with corrected titles
[Zimm4PlotsV1.pptx](https://github.com/groops-devs/groops/files/9243457/Zimm4PlotsV1.pptx)

**Reply by Waley04** [reply]
> updated version with corrected titles [Zimm4PlotsV1.pptx](https://github.com/groops-devs/groops/files/9243457/Zimm4PlotsV1.pptx)

Hi Jean,

Let me send an example of this shortly. It seems something is wrong with your results.

Regards,

**Comment by Bemtgenj** [comment]
Hi Waley

txs for sending - i create a dropbox so we can send the big files back and forth if you send me your email address

Cheers
Jean

**Comment by Waley04** [comment]
Hi Jean, 

I have problems with my Google Drive. Instead, I managed to upload the VFM (**vmf3_grid_2006-08.dat.gz**) file [here](https://github.com/Waley04/miscellaneousGnssIssues/blob/main/vmf3_grid_2006-08.dat.gz). I am therefore wondering if you could try to process using this file. 

With it, you should be able to obtain similar results to mine.

Best wishes,

Waley.

**Comment by Bemtgenj** [comment]
Hi Waley,

Thank you so much for managing to send us this file (i just downloaded it). A bit of of updates form our side: so far we have been running GROOPS on 2 Linux machines and one windows PC. Up to now all 3 PC showed the same "messy" results. Yesterday our supervisor installed  the latest version on his Linux PC and got the same results as you (using the same Vienna Mapping script), we then decided to upgrade to the latest version - 1 Linux PC is now updated, one Windows PC in the process of update, as we still have one PC with the old version (that showed the same "messy" results, we will run your vmf3.. on that machine and se what type of results we will get with your version of  the file. We are now thinking it might a a complier problem of the Groops installation... but then the strange thing is that it is on 3 machines.....

i will keep you posted.

Cheers and txs again for all the help
Jean

**Comment by Bemtgenj** [comment]
Hi Waley,

As mentioned above we ran you vmf file on the Linux PC -same version as before / not updated to latest version and get the results show the same plots as we had in the past  - this would mean that it is not related to the  vmf mapping script / file. as the new version does not seem to have this issue on our Pcs we will also upgrade that one.

Again, txs very much for your feedback and help with this issue

Cheers
Jean

![position_2006-08-02 zimm](https://user-images.githubusercontent.com/100558965/182868781-14c1e042-74a8-44c8-8b3a-79b0cf94e695.png)

**Comment by Waley04** [comment]
Hi Jean,

In my case, I have multiple versions of GROOPS on different operating systems. I get good results on all. 

As long as you compiled the software successfully, then the results should be okay. In other words, the results may not be related to compilation issues. In my case, I have the following versions on one working station:

**1).** **20220426**

![01groopsApp](https://user-images.githubusercontent.com/101952777/182888962-622fe3a1-d88b-47ba-883d-7ab3301f7242.PNG)

**2).** **20220531**
![02groopsApp](https://user-images.githubusercontent.com/101952777/182889066-0ac22857-8b65-4003-9d3e-a6ada990476e.PNG)

**3).** **20220728**
![03groopsApp](https://user-images.githubusercontent.com/101952777/182889207-b351e395-03ad-454f-baa2-adb503908cdc.PNG)

I also have the releases on my other Linux and Windows PC:

**Groops: 2021-02-02**
==> [2021-02-02](https://github.com/groops-devs/groops/releases/tag/2021-02-02)
**Groops: 2020-11-12**
===> [2020-11-12](https://github.com/groops-devs/groops/releases/tag/2020-11-12)
 
Note: Using the same data, I get the results I showed last time. 

Thanks with regards,

Waley.

**Comment by Bemtgenj** [comment]
Hi Waley

Indeed that is strange that we have the issue and you don' t see it on any versions you run. the version we used ( on all 3 machines was the one from September last year - 6Sep2021)
 I am still not sure where the problem came from - if that version had some kind of bug - don't think so, but not sure) 

Cheers
Jean

PS struggling a bit with installation of the new version on the windows machine - the scripts don't run with groopsgui and when i just run groops.exe (to see what is does) i get an procedure entry could not be located ... message (libidn2-0.dll)  / works fine on the linux machines

**Comment by Bemtgenj** [comment]
Update on the installation - works now after i downloaded the libidin2-0.dll and replaced the one on the PC

**Comment by zhedumi** [comment]
Hi I am a bit late to the discussion but just to be sure I downloaded the public available [PPP script](https://ftp.tugraz.at/outgoing/ITSG/groops/) and did the computations for zimm with the TUG products and your observation file on a few different groops version. My results are nearly exactly the same as @Waley04 . They differences that still exist is probably some different outlier detection results and weighting.

![position_2006-08-02 zimm](https://user-images.githubusercontent.com/77068736/183046121-8637ebe4-8457-4ef5-aeeb-8e98377c792a.png)

@Bemtgenj  within the [Data.zip](https://github.com/groops-devs/groops/files/9195958/Data.zip) the attitude files (Orbex files) were missing but I presume you still used our files there. Is the problem still occuring or did you manage to fix it with @Waley04  suggestions?
Can you by any chance provide the log file from the groops output? If you start it by console you just need to add the commandline parameter -l and a name for the logFile. 
Between your script and my script I do not see that much difference the only thing that is different might be the vmf3_grid. Here is a link to mine in case you want to test it: [vmf3_grid_2006-08.dat.gz](https://www.dropbox.com/s/0kkbarbkuyhxf95/vmf3_grid_2006-08.dat.gz?dl=0)

**Comment by Bemtgenj** [comment]
Hi Patrick

txs for the feedback , this is as well puzzling to us , we have used vmf3 and v3gr files as input to the Vienna mapping script and got very similar "strange " result for all runs (i did a 2009 aug 2 and had a similar effects however aug 2 2020 was fine) 

Attached iare differnt log files: WSVMF = waleyscriptVienna Mapping / vmf3 input file - GROOPSVMF3 = Groops Vienna mapping script from Sebastian/ vmf3 input file, -> all showed the same effects around 12:00pm. As mentioned we had this on 3 different machines 2 Linux / 1 windows), even with the gpt model instead of the full Vienna mapping

i will not be able to run them easily as this morning i upgraded to the latest version
Cheers
Jean


[temp.zip](https://github.com/groops-devs/groops/files/9267690/temp.zip)

**Comment by zhedumi** [comment]
```

2022-08-02 15:05:51 Status  1. iteration  --------------------------

2022-08-02 15:05:51 Status  accumulate normals
2022-08-02 15:05:53 Status  2881 loops in time 00:00:02
2022-08-02 15:05:53 Status  constrain constraint.troposphere (31 parameters)
2022-08-02 15:05:53 Status  solve
2022-08-02 15:05:55 Status  2881 loops in time 00:00:02
2022-08-02 15:05:55 Info      **sigma = 19.81**
2022-08-02 15:05:55 Status  Compute residuals
2022-08-02 15:05:57 Status  2881 loops in time 00:00:02
2022-08-02 15:05:57 Status  Downweight outliers
2022-08-02 15:05:57 Status  Estimate variance factors
2022-08-02 15:05:57 Info      C1CG**: sigma0 = 1.00, redundancy = 1.00, count =   19593, outliers =    742 (3.79 %)
2022-08-02 15:05:57 Info      C2WG**: sigma0 = 1.00, redundancy = 1.00, count =   19593, outliers =    892 (4.55 %)
**2022-08-02 15:05:57 Info      L1?G**: sigma0 = 1.04, redundancy = 0.30, count =   19593, outliers =   2640 (13.47 %)
2022-08-02 15:05:57 Info      L2?G**: sigma0 = 1.04, redundancy = 0.11, count =   19593, outliers =   2640 (13.47 %)**

```
Yeah in the logFiles it shows definitly an issue. The sigmaFactor is way to high with 19.81 even for a first iteration. Even the number of outliers is way to high. 
Something I also do not fully comprehend yet is that you have less number of observations and parameters opposed to my output while we both use the same observations and inputData. I will need to look at your preprocessing script if there is something at fault so I give me a bit more time to check what is wrong.

From your logFiles:
```
2022-08-02 15:05:49 Status  1 loops in time 00:00:01
2022-08-02 15:05:49 Info      0 disabled stations
2022-08-02 15:05:49 Info      transmitter: 29
2022-08-02 15:05:49 Info      receiver:    1
2022-08-02 15:05:49 Info    types and number of observations:
2022-08-02 15:05:49 Info      C1CG**:     19593
2022-08-02 15:05:49 Info      C2WG**:     19593
2022-08-02 15:05:49 Info      L1?G**:     19593
2022-08-02 15:05:49 Info      L2?G**:     19593
2022-08-02 15:05:49 Info            + =========
2022-08-02 15:05:49 Info      total:      78372
2022-08-02 15:05:49 Info      number of tracks: 58
2022-08-02 15:05:49 Status  setup parameters
2022-08-02 15:05:49 Info        19593 constrained STEC parameters (preeliminated)
2022-08-02 15:05:49 Info         2878 VTEC epoch parameters
2022-08-02 15:05:49 Info         2878 receiver clock error parameters
2022-08-02 15:05:49 Info         8634 kinematic position parameters
2022-08-02 15:05:49 Info           25 troposphere wet parameters
2022-08-02 15:05:49 Info            6 troposphere gradient parameters
2022-08-02 15:05:49 Info            1 receiver TEC bias parameters
2022-08-02 15:05:49 Info            2 receiver phase bias parameters
2022-08-02 15:05:49 Info          114 ambiguity parameters
2022-08-02 15:05:49 Info    + =======
```

This is from my logfile.
```
2022-08-05 11:12:56 Status  (process  4) 1 loops in time 00:00:01
2022-08-05 11:12:56 Info    (process  4)   transmitter: 29
2022-08-05 11:12:56 Info    (process  4)   receiver:    1
2022-08-05 11:12:56 Info    (process  4) types and number of observations:
2022-08-05 11:12:56 Info    (process  4)   C1CG**:     23395
2022-08-05 11:12:56 Info    (process  4)   C2WG**:     23395
2022-08-05 11:12:56 Info    (process  4)   L1?G**:     23395
2022-08-05 11:12:56 Info    (process  4)   L2?G**:     23395
2022-08-05 11:12:56 Info    (process  4)         + =========
2022-08-05 11:12:56 Info    (process  4)   total:      93580
2022-08-05 11:12:56 Info    (process  4)   number of tracks: 60
2022-08-05 11:12:56 Status  (process  4) setup parameters
2022-08-05 11:12:56 Info    (process  4)     23395 constrained STEC parameters (preeliminated)
2022-08-05 11:12:56 Info    (process  4)      2878 VTEC epoch parameters
2022-08-05 11:12:56 Info    (process  4)      2878 receiver clock error parameters
2022-08-05 11:12:56 Info    (process  4)      8634 kinematic position parameters
2022-08-05 11:12:56 Info    (process  4)        13 troposphere wet parameters
2022-08-05 11:12:56 Info    (process  4)         4 troposphere gradient parameters
2022-08-05 11:12:56 Info    (process  4)         1 receiver TEC bias parameters
2022-08-05 11:12:56 Info    (process  4)         2 receiver phase bias parameters
2022-08-05 11:12:56 Info    (process  4)       118 ambiguity parameters
```

But with the new version it is fixed now? Do you also have a logFile from a computation with a new groops version?

**Comment by Bemtgenj** [comment]
HI Patrick

Here are the two convertsat scripts (Star Camera / Orbex, I have not yet run the new script as i was now trying to dix the QT part of the problem i had with the  installation. I can ask Parisa to upload  it (she used to have the same problem and with the new version it is fixed)
Cheers
Jean

[temp.zip](https://github.com/groops-devs/groops/files/9268082/temp.zip)
)

**Reply by zhedumi** [reply]
No need to stress yourself for the files it will take some time for me to go through all of the preprocessing anyway.

**Comment by ParisaShafiei** [comment]
Hi Patrick, 

As Jean has already mentioned, when we installed the latest version of GROOPS on our systems, it is working properly. Indeed, we assume that there was something wrong with the previous version, i.e. 6 Sep 2021, that we could not get the proper results. 

Here, I have attached the the scripts as well as the log files for the ZIMM station. I ran it on my linux machine. 

Thank you in advance for your time and help. 

With Kind Regards, 
Parisa 

[ZIMM-Scripts.zip](https://github.com/groops-devs/groops/files/9268883/ZIMM-Scripts.zip)

**Comment by Waley04** [comment]
Hi @Parisa

In my case, I have the same results for all versions including the following releases:

==> [Groops Version: 2021-02-02](https://github.com/groops-devs/groops/releases/tag/2021-02-02)
==> [Groops Version: 2021-09-06](https://github.com/groops-devs/groops/releases/tag/2021-09-06)

Thanks with best regards,

Waley.

**Reply by ParisaShafiei** [reply]
Hi Walely, 

Thanks for informing us. We have exactly used the same version for September 2021. To be honest, now I have no idea why the processing did not work and could not manage to get your outputs. However, with the latest version, everything is fine and it's working properly for Jean and I. 

We need to wait for the answer's of the GROOPS's developers. 

With Kind Regards, 
Parisa
